monai/networks/layers/simplelayers.py

```diff
@@ -184,23 +195,106 @@ def separable_filtering(x: torch.Tensor, kernels: Union[Sequence[torch.Tensor],
         for s in ensure_tuple_rep(kernels, spatial_dims)
     ]
     _paddings = [cast(int, (same_padding(k.shape[0]))) for k in _kernels]
-    n_chns = x.shape[1]
+    n_chs = x.shape[1]
 
     def _conv(input_: torch.Tensor, d: int) -> torch.Tensor:
         if d < 0:
             return input_
         s = [1] * len(input_.shape)
         s[d + 2] = -1
         _kernel = kernels[d].reshape(s)
-        _kernel = _kernel.repeat([n_chns, 1] + [1] * spatial_dims)
+        # if filter kernel is unity, don't convolve
+        if torch.equal(_kernel.squeeze(), torch.ones(1, device=_kernel.device)):
+            return _conv(input_, d - 1)
+        _kernel = _kernel.repeat([n_chs, 1] + [1] * spatial_dims)
         _padding = [0] * spatial_dims
         _padding[d] = _paddings[d]
-        conv_type = [F.conv1d, F.conv2d, F.conv3d][spatial_dims - 1]
-        return conv_type(input=_conv(input_, d - 1), weight=_kernel, padding=_padding, groups=n_chns)
+
+        if mode == "zeros":  # if zero padding (default), can use functional convolution
+            conv_type = [F.conv1d, F.conv2d, F.conv3d][spatial_dims - 1]
+            return conv_type(input=_conv(input_, d - 1), weight=_kernel, padding=_padding, groups=n_chs)
+        else:
+            _k_len = _kernel.shape.numel()
+            _pad_tup = tuple(_padding)
+            convolver = conv_factory(spatial_dims)(
```

Comment:
`conv_factory` usage is something like:
```py
from monai.networks.layers.factories import Conv

Conv[Conv.CONV, spatial_dims]
```

also why not merge the functional case into the nn.Conv* case?

Reply:
I had some trouble getting the `nn.Conv*` classes to work for all tests, so am now using `nn.functional.pad` to pad the Tensors directly, allowing use of `nn.functional.conv*`. This has added benefit of working with PyTorch versions before 1.5.1, so I have also removed all the version checking.